internal/services/storage/storage_account_resource.go

```diff
@@ -223,6 +227,39 @@ func resourceStorageAccount() *pluginsdk.Resource {
 				},
 			},
 
+			"customer_managed_key": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"key_vault_id": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: keyVaultValidate.VaultID,
+						},
+
+						"key_name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"key_version": {
```

Comment:
could we instead use keyvault key id for all three of these properties?

Reply:
Initially tried that with a schema like below to still allow for auto rotation of the key (it's set by omitting the `key_version`). But it would mean i'd need to add a DiffSuppress to `key_vault_key_id` since I might only be able to generate a versionless ID for the key from the information returned by the API.
```
	Schema: map[string]*pluginsdk.Schema{
		"key_vault_key_id": {
			Type:         pluginsdk.TypeString,
			Required:     true,
			ValidateFunc: keyVaultValidate.NestedItemId,
		},
		"auto_key_rotation_enabled": {
			Type:         pluginsdk.TypeBool,
			Optional:     true,
			Default:      false
		},
```

I could do something like `key_vault_key_versionless_id` to only replace `key_vault_id` and `key_name`, would that be an appropriate alternative?